src/main/java/spoon/SpoonException.java

```diff
@@ -21,4 +28,19 @@ public SpoonException(Throwable e) {
 	public SpoonException(String msg, Throwable e) {
 		super(msg, e);
 	}
+	/**
+	 * Handles a JLSViolation according to the environment settings. If {@link Environment#getIgnoreSyntaxErrors()} is set to true, the exception is ignored.
+	 * Otherwise, the exception is thrown.
+	 * @param holder  an object with access to factory and environment.
+	 * @param reason  the reason for the exception.
+	 */
+	public static void handleJLSViolation(FactoryAccessor holder, String reason) {
+		if (holder != null && holder.getFactory() != null
+				&& holder.getFactory().getEnvironment() != null
+				&& !holder.getFactory().getEnvironment().getIgnoreSyntaxErrors()) {
+			throw new JLSViolation(reason);
+		} else {
+			LOGGER.info("An element is not compliant to the JLS. See: {}", reason);
+		}
```

Comment:
Here we could say ignoring means no logging at all. Any opinions on just log it, even if the user said `gnoreSyntaxErrors`?

Reply:
It's fine to log IMO.